cli/smoke.go

```diff
@@ -65,90 +69,65 @@ func (c *StartEvergreenCommand) Execute(_ []string) error {
 	if c.Conf == "" {
 		c.Conf = filepath.Join(wd, "scripts", defaultConfigFile)
 	}
-	web := exec.Command(c.Binary, "service", "web", "--conf", c.Conf)
-	web.Env = []string{fmt.Sprintf("EVGHOME=%s", wd), "PATH=" + strings.Replace(os.Getenv("PATH"), `\`, `\\`, -1)}
-	webSender := send.NewWriterSender(send.MakeNative())
-	defer webSender.Close()
-	webSender.SetName("web.service")
-	web.Stdout = webSender
-	web.Stderr = webSender
-
-	runner := exec.Command(c.Binary, "service", "runner", "--conf", c.Conf)
-	runner.Env = []string{fmt.Sprintf("EVGHOME=%s", wd), "PATH=" + strings.Replace(os.Getenv("PATH"), `\`, `\\`, -1)}
-	runnerSender := send.NewWriterSender(send.MakeNative())
-	defer runnerSender.Close()
-	runnerSender.SetName("runner")
-	runner.Stdout = runnerSender
-	runner.Stderr = runnerSender
-
-	if err = web.Start(); err != nil {
-		return errors.Wrap(err, "error starting web service")
-	}
-	if err = runner.Start(); err != nil {
-		return errors.Wrap(err, "error starting runner")
+	if !c.Runner && !c.Web {
+		return errors.New("Must specify --web or --runner (or both)")
 	}
 
-	exit := make(chan error)
-	interrupt := make(chan os.Signal, 1)
-	signal.Notify(interrupt, os.Interrupt)
-	go func() {
-		exit <- web.Wait()
-		grip.Errorf("web service exited: %s", err)
-	}()
-	go func() {
-		exit <- runner.Wait()
-		grip.Errorf("runner exited: %s", err)
-	}()
-
-	select {
-	case <-exit:
-		err = errors.New("problem running Evergreen")
-		grip.Error(err)
-		return err
-	case <-interrupt:
-		grip.Info("received SIGINT, killing Evergreen")
-		if err := web.Process.Kill(); err != nil {
-			grip.Errorf("error killing evergreen web service: %s", err)
+	exit := make(chan error, 2)
+	if c.Web {
+		web := exec.Command(c.Binary, "service", "web", "--conf", c.Conf)
+		web.Env = []string{fmt.Sprintf("EVGHOME=%s", wd), "PATH=" + strings.Replace(os.Getenv("PATH"), `\`, `\\`, -1), "EVR_TASK_ID=" + os.Getenv("EVR_TASK_ID"), "EVR_AGENT_PID" + os.Getenv("EVR_AGENT_PID")}
```

Comment:
shouldn't we just do something like: 

web.Env = append(os.Environ(), "EVGHOME=foo")

Reply:
done